Automattic/jetpack
# Search: Refactor free plan checking and remove disabling Module Control on Paid plan

Fixes #26844 

#### Changes proposed in this Pull Request:

* Refactor the free plan checking by `effective_subscription?.product_slug`.
* Introduce `isDisabledFromOverLimitOnFreePlan` to allow paid users to keep operating even over the limit.

#### Other information:


#### Jetpack product discussion


No

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Purchase the Jetpack Search `Free` plan.
* Navigate to the site with URL `/wp-admin/admin.php?page=jetpack-search&new_pricing_202208=1`.
* Append ` || true` to `plan_usage.must_upgrade` to mock `isOverLimit`.
* Ensure the Module Control is disabled under an overlay.
* Upgrade to a paid plan.
* Ensure the Module Control is operable.
